vmaas/reposcan/database/csaf_store.py

```diff
@@ -28,151 +56,327 @@ def delete_csaf_file(self, name: str):
             CSAF_FAILED_DELETE.inc()
             self.logger.exception("Failed to delete csaf file.")
             self.conn.rollback()
+            raise CsafStoreException("Failure while deleting CSAF file")
         finally:
             cur.close()
 
-    def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
+    def _save_csaf_files(self, csaf_files: model.CsafFiles) -> None:
         cur = self.conn.cursor()
-        db_ids = execute_values(
-            cur,
+        try:
+            rows = execute_values(
+                cur,
+                """
+                    insert into csaf_file (name, updated) values %s
+                    on conflict (name) do update set updated = excluded.updated
+                    returning id, name
+                """,
+                csaf_files.to_tuples(("name", "csv_timestamp")),
+                fetch=True,
+            )
+            self.conn.commit()
+        except Exception as exc:
+            CSAF_FAILED_IMPORT.inc()
+            self.logger.exception("Failed to import csaf file to DB: %s", exc)
+            raise CsafStoreException("Failed to import csaf file to DB")
+        finally:
+            cur.close()
+        for row in rows:
+            # translate file name to cve, e.g. 2023/cve-2023-0030.json -> CVE-2023-0030
+            match = re.search(CVE_PATTERN, row[1])
+            if match:
```

Comment:
What happens when there is a valid CVE file in repo which name doesn't match? I'd not rely on file name, rather extract the CVE name from inside the file.